hibernate-core/src/main/java/org/hibernate/sql/ast/spi/AbstractSqlAstTranslator.java

```diff
@@ -1092,6 +1095,59 @@ protected Predicate determineWhereClauseRestrictionWithJoinEmulation(AbstractUpd
 		querySpec.getSelectClause().addSqlSelection(
 				new SqlSelectionImpl( new QueryLiteral<>( 1, getIntegerType() ) )
 		);
+
+		final MutableObject<StandardVirtualTableGroup> virtualTableGroup;
+		final List<TableGroupJoin> collectedNonInnerJoins;
+		if ( supportsJoinInMutationStatementSubquery() ) {
+			virtualTableGroup = new MutableObject<>();
+			collectedNonInnerJoins = new ArrayList<>();
+			emulateWhereClauseRestrictionJoins( statement, querySpec, tableGroupJoin -> {
+				if ( tableGroupJoin.getJoinType() == SqlAstJoinType.INNER ) {
+					final TableGroup joinedGroup = tableGroupJoin.getJoinedGroup();
+					if ( !virtualTableGroup.isSet() ) {
+						virtualTableGroup.set( new StandardVirtualTableGroup(
+								joinedGroup.getNavigablePath(),
+								joinedGroup.getModelPart(),
+								joinedGroup,
+								joinedGroup.isFetched()
+						) );
+					}
+					virtualTableGroup.get().addTableGroupJoin( tableGroupJoin );
+					querySpec.applyPredicate( tableGroupJoin.getPredicate() );
```

Comment:
Shouldn't this rather be the following?

```suggestion
					final FromClause fromClause = querySpec.getFromClause();
					if ( fromClause.getRoots().isEmpty() ) {
						final TableGroup joinedGroup = tableGroupJoin.getJoinedGroup();
						fromClause.addRoot( new StandardVirtualTableGroup(
								joinedGroup.getNavigablePath(),
								joinedGroup.getModelPart(),
								joinedGroup,
								joinedGroup.isFetched()
						) );
						querySpec.applyPredicate( tableGroupJoin.getPredicate() );
					}
					else {
						fromClause.getRoots().get( 0 ).addTableGroupJoin( tableGroupJoin );
					}
```